edweiss-app/components/todo/todoDisplay.tsx

```diff
@@ -0,0 +1,273 @@
+     * Note: this function manages the distinction between a
+     * vertical/horizontal scroll
+     */
+    const onHandlerStateChange = (event: any) => {
+        const { state, translationY, translationX } = event.nativeEvent;
+
+        // Note: Record the starting position of the gesture
+        if (state === State.BEGAN) {
+            gestureStartX.current = event.nativeEvent.x;
+            gestureStartY.current = event.nativeEvent.y;
+            isEdgeSwipe.current = (gestureStartX.current <= edgeZoneWidth ||
+                gestureStartX.current >= screenWidth - edgeZoneWidth);
+        }
+
+        /* Note: Check if the user started the gesture near the 
+         * edges and privilledge the vertical scroll            */
+        if (state === State.ACTIVE) {
+            if (isEdgeSwipe.current) {
+                if (Math.abs(translationY) > Math.abs(translationX)) {
+                    return;
+                }
+                translateX.setValue(translationX);
+            }
+        }
+
+        // Note: Handle snapping back or going to the edit screen
+        if (state === State.END) {
+            handleGestureEnd(event);
+        }
+    };
+
+
+
+    return (
+
+        <>
+            <TView >
+
+                {/* Background Archive Text */}
+                <BackgroundArchiveText todo={todo} />
+
+
+                {/* Front Card Display */}
+                <PanGestureHandler
+                    onGestureEvent={handlePanGestureEvent}
+                    onHandlerStateChange={onHandlerStateChange}
+                >
+
+
+                    {/* The Animated TODO Card */}
+                    <Animated.View
+                        style={{
+                            transform: [{ translateX }],
+                            borderRadius: 20,
+                            backgroundColor: bgColor,
+                            margin: 13,
+                            overflow: 'hidden',
+                            zIndex: 100
+                        }}
+                    >
+                        <TTouchableOpacity
+                            b={'lg'}
+                            m='sm'
+                            p='lg'
+                            backgroundColor='base'
+                            borderColor='surface0'
+                            radius='lg'
+                        >
+                            <TView flexDirection='row' justifyContent='space-between'>
+                                <TView flex={1} mr='md'>
+                                    <TTouchableOpacity
+                                        // Display the modal on simple press
+                                        onPress={() => { setTodoToDisplay(todo); modalRef.current?.present(); }}
+
+                                        // Navigate to edit screen on long press
+                                        onLongPress={() => {
+                                            Vibration.vibrate(100);
```

Comment:
Suggestion : You currently use vibration for long presses; consider allowing users to toggle this feature in settings for enhanced accessibility.